ssh_key_rotator/server/data_stores.py

```diff
@@ -63,18 +64,35 @@ class FileSystemDataStore(DataStore):
 
     def __init__(self, temp: bool = False):
         self.temp = temp
-        self.dir = f"{get_user_path()}/.ssh"
+        home_dir = f"{get_user_path()}/.ssh"
+        if temp:
+            self._dir: str | TemporaryDirectory = (
+                TemporaryDirectory(  # pylint: disable=consider-using-with
+                    prefix="ssh-keys-", dir=home_dir
+                )
+            )
+        else:
+            self._dir = home_dir
+
+    @property
+    def dir(self) -> str:
+        if isinstance(self._dir, str):
+            return self._dir
+        return self._dir.name
 
     def get_sshd_config_line(self) -> str:
-        return "local"
+        return f"local {self.dir}"
 
     def delete_key(self, host: str, user: str):
-        os.remove(f"{get_user_path()}/.ssh/{host}/{user}")
+        os.remove(f"{self.dir}/{host}/{user}")
 
     def __enter__(self):
-        if self.temp and not os.path.isdir(self.dir):
```

Comment:
simply `mkdirs` and deal with the exception, if any (remember that testing for something and then acting on it is always a race - races are not good in code)